owncloudApp/src/main/java/com/owncloud/android/authentication/AccountAuthenticator.java

```diff
@@ -316,30 +318,40 @@ private void refreshToken(
         Timber.d("Get OAuth2 refresh token from account: %s, to exchange it for new access and refresh tokens",
                 refreshToken);
 
-        String baseUrl = accountManager.getUserData(
-                account,
-                AccountUtils.Constants.KEY_OC_BASE_URL
-        );
+        AuthStateManager authStateManager = AuthStateManager.getInstance(mContext);
+        AuthorizationServiceConfiguration authorizationServiceConfiguration = authStateManager.readState(account.name).
+                getAuthorizationServiceConfiguration();
+
+        if (authorizationServiceConfiguration == null) {
+            Timber.d("No authorization configuration found, falling back to hardcoded oauth2 endpoints");
+            // The code below is for users (already logged in) updating the app from previous versions, which do not have
+            // an authState that is configured when doing a fresh log in
+            String baseUrl = accountManager.getUserData(
+                    account,
+                    AccountUtils.Constants.KEY_OC_BASE_URL
+            );
+            authorizationServiceConfiguration = new AuthorizationServiceConfiguration(
+                    Uri.parse(baseUrl + "/" + mContext.getString(R.string.oauth2_url_endpoint_auth)), // auth endpoint
+                    Uri.parse(baseUrl + "/" + mContext.getString(R.string.oauth2_url_endpoint_access)) // token endpoint
```

Comment:
In OAuthUtils you use PATH_SEPARATOR, why don't you use it here?

Reply:
👍